transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/tworegion/TwoRegionEndToEndRetrievalTests.groovy

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © 2013-2014 The Hyve B.V.
+ *
+ * This file is part of transmart-core-db.
+ *
+ * Transmart-core-db is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * transmart-core-db.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.transmartproject.db.dataquery.highdim.tworegion
+
+import com.google.common.collect.Lists
+import grails.test.mixin.TestMixin
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.transmartproject.core.dataquery.highdim.HighDimensionDataTypeResource
+import org.transmartproject.core.dataquery.highdim.HighDimensionResource
+import org.transmartproject.core.dataquery.highdim.assayconstraints.AssayConstraint
+import org.transmartproject.core.dataquery.highdim.dataconstraints.DataConstraint
+import org.transmartproject.core.dataquery.highdim.projections.Projection
+import org.transmartproject.db.test.RuleBasedIntegrationTestMixin
+
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.*
+
+/**
+ * Created by j.hudecek on 17-3-14.
+ */
+@TestMixin(RuleBasedIntegrationTestMixin)
+class TwoRegionEndToEndRetrievalTests {
+
+    HighDimensionResource highDimensionResourceService
+
+    HighDimensionDataTypeResource resource
+
+    Closeable dataQueryResult
+
+    TwoRegionTestData testData = new TwoRegionTestData()
+
+    AssayConstraint trialNameConstraint
+
+    @Before
+    void setUp() {
+        testData.saveAll()
+
+        resource = highDimensionResourceService.getSubResourceForType 'two_region'
+        assertThat resource, is(notNullValue())
+
+        trialNameConstraint = resource.createAssayConstraint(
+                AssayConstraint.TRIAL_NAME_CONSTRAINT,
+                name: TwoRegionTestData.TRIAL_NAME)
+    }
+
+    @After
+    void after() {
+        dataQueryResult?.close()
+    }
+
+    @Test
+    void basicTestWithConstraints() {
+        List dataConstraints = [resource.createDataConstraint(
+                [chromosome: "1", start: 6, end: 12], DataConstraint.DISJUNCTION_CONSTRAINT
```

Comment:
I think you meant another constraint here. It's surprising it works at all (the disjunction constraint factory should give an error). Plus, you can use groovy syntax, with collects maps entries to a first parameter map (`resource.createDataConstraint('chromosome_segment' /* or whatever */, chromosome: "1", start: 6, end:12)`),